### PR TITLE
fix(compiler): implicit await missing in argument lists

### LINKED ISSUE #1073: await is not called inside functions
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### I tried this:

I tried compiling the following wing code
```ts (wing)
bring cloud;

let q = new cloud.Queue();
let str_to_str = new cloud.Function(inflight (s: str): str => {

}) as "str_to_str";
let func = new cloud.Function(inflight (s: str): str => {
  str_to_str.invoke("one");
  print(str_to_str.invoke("two"));
}) as "func";
```

### I expected this:

await to be called on for both invocations of `str_to_str.invoke`

### Instead, this happened:

The output of the `func` is (in the playground) is:

```ts
async handle(s) { const { str_to_str } = this; {
  (await str_to_str.invoke("one"));
  {console.log((str_to_str.invoke("two")))};
} };
```

Notice that we should `await` the two calls for `str_to_str.invoke`

### Component:

Compiler

### Wing Version:

0.4.61

### Node.js Version:

_No response_

### Platform(s):

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by staycoolcall911**
See issue #1073 for the use case - in short, we didn't propagate the phase (e.g. `inflight`) when jsifying arg list, which led to skipping adding implicit await on inflight calls.

Credit for assistance: @yoav-steinberg 👍 

Fixes #1073 

*By submitting this pull request, I confirm that my contribution is made under the terms of the 
[Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingc/src/jsify.rs [hunk 1]
```diff
@@ -198,7 +208,7 @@ impl JSifier {
 		}
 
 		for arg in arg_list.pos_args.iter() {
-			args.push(self.jsify_expression(arg, Phase::Independent));
+			args.push(self.jsify_expression(arg, phase));
```
Comment (eladb) [inline libs/wingc/src/jsify.rs:211]: Well, well, well...

https://github.com/winglang/wing/pull/986/files#r1056930513

**Review by eladb — APPROVED** [review]
Nice.

**Comment by eladb** [comment]
The title needs some work - for bugs, the title should reflect the problem, not the solution.
So maybe something like:

```
fix(compiler): implicit await missing in argument lists
```

File: libs/wingc/src/jsify.rs (hunk 1 above)
Comment (staycoolcall911) [reply libs/wingc/src/jsify.rs:211]: 🤷‍♂️

**Review by yoav-steinberg — APPROVED** [review]

File: tools/hangar/src/cli.test.ts [hunk 2]
```diff
@@ -138,7 +138,7 @@ test.each(validWingFiles)(
       expect(npx_tfJson).toMatchSnapshot("cdk.tf.json");
 
       // get all files in .wing dir
-      const dotWingFiles = await walk.async(path.join(targetDir, ".wing"), {
+      const dotWingFiles = await walk.sync(path.join(targetDir, ".wing"), {
```
Comment (Chriscbr) [inline tools/hangar/src/cli.test.ts:141]: 🤯  What a clean fix @staycoolcall911

Comment (staycoolcall911) [reply tools/hangar/src/cli.test.ts:141]: Yeah, but it didn't work :)
It's not hangar's fault that our snapshot tests are printing inflight files in a non-deterministic order. That actually happens before, either on our compiler or SDK. I'm trying to look into it, but would love pointers

**Pull request merged**